src/main/java/edu/hm/hafner/analysis/parser/ProtoLintJsonParser.java

```diff
@@ -0,0 +1,88 @@
+package edu.hm.hafner.analysis.parser;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import edu.hm.hafner.analysis.Issue;
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.ParsingCanceledException;
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.ReaderFactory;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+
+/**
+ * Parser for ProtoLint reports.
+ *
+ * <p>The recommended report format is JSON. The JSON report contains more information (affected rule, severity,
+ * basedir) as the plaintext format. For full feature set please use protoLint &gt;= 0.50.2.
+ *
+ * <p>We still support plaintext reports and JSON reports generated with protoLint &lt; 0.50.2 as well.
+ *
+ * @author David Hart
+ * @author github@profhenry.de
+ * @see <a href="https://github.com/yoheimuta/protolint">https://github.com/yoheimuta/protolint</a>
+ */
+public class ProtoLintJsonParser extends JsonIssueParser {
+    private static final long serialVersionUID = 573706779074579673L;
+
+    /**
+     * Old plaintext parser used as fallback.
+     */
+    private final ProtoLintParser plainParser = new ProtoLintParser();
+
+    @Override
+    public boolean accepts(ReaderFactory aReaderFactory) {
+        return true;
+    }
+
+    @Override
+    public Report parse(final ReaderFactory aReaderFactory) throws ParsingException, ParsingCanceledException {
+        // We try parsing the JSON report
+        if (super.accepts(aReaderFactory)) {
+            return super.parse(aReaderFactory);
+        }
+        // and fall back to plaintext report
+        return plainParser.parse(aReaderFactory);
+    }
```

Comment:
This somehow duplicates the code of the `CompositeParser`, can't you make use of this standard concept when using the results of multiple parsers?

Reply:
Now using composite parser